Backend/app/rider-platform/rider-app/Main/src/API/UI/Issue.hs

```diff
@@ -97,17 +105,26 @@ castMOCityById :: (CacheFlow m r, EsqDBFlow m r, EsqDBReplicaFlow m r) => Id Com
 castMOCityById moCityId = do
   moCity <- CQMOC.findById (cast moCityId)
   return $ fmap castMOCity moCity
-  where
-    castMOCity moCity =
-      Common.MerchantOperatingCity
-        { id = cast moCity.id,
-          merchantId = cast moCity.merchantId,
-          city = moCity.city
-        }
+
+castMOCityByMerchantShortIdAndCity :: (CacheFlow m r, EsqDBFlow m r) => ShortId Common.Merchant -> Context.City -> m (Maybe Common.MerchantOperatingCity)
+castMOCityByMerchantShortIdAndCity (ShortId merchantShortId) opCity = do
+  merchantOpCity <- CQMOC.findByMerchantShortIdAndCity (ShortId merchantShortId) opCity
+  return $ fmap castMOCity merchantOpCity
+
+castMOCity :: DMOC.MerchantOperatingCity -> Common.MerchantOperatingCity
+castMOCity moCity =
+  Common.MerchantOperatingCity
+    { id = cast moCity.id,
+      city = moCity.city,
+      merchantId = cast moCity.merchantId
+    }
 
 castRideInfo :: Id Common.Merchant -> Id Common.MerchantOperatingCity -> Id Common.Ride -> Flow Common.RideInfoRes
 castRideInfo merchantId _ rideId = do
-  rideInfoRes <- DRide.rideInfo (cast merchantId) (cast rideId)
+  rideInfoRes <-
+    Redis.safeGet makeRideInfoCacheKey >>= \case
```

Comment:
Why caching?

Reply:
Too heavy as multiple queries are being used to make RideInfoRes and optionsApi is called multiple times, so caching is required for 3 days (max duration in which issue can be reported for a ride)

Reply:
Currently, we only allow Issue Reports in case if ride is completed or cancelled. So status shouldn't be a problem in that case. But if we want to allow reports for ongoing rides then we would need to cache this info only in case if status is terminal.

Reply:
Yeah can you add that check too.

Reply:
Added Check